quarkus/dist/src/main/content/bin/kc.sh

```diff
@@ -55,9 +56,24 @@ do
     case "$1" in
       --debug)
           DEBUG_MODE=true
-          if [ -n "$2" ] && expr "$2" : '[0-9]\{0,\}$' >/dev/null; then
-              DEBUG_PORT=$2
-              shift
+          if [ -n "$2" ]; then
+              # Plain port
+              if echo "$2" | grep -Eq '^[0-9]+$'; then
+                  DEBUG_ADDRESS="$2"
+                  shift
+              # Bare IPv6 literal (no port)
+              elif echo "$2" | grep -Eq '^[0-9A-Fa-f:]+$'; then
+                  DEBUG_ADDRESS="[$2]:$DEBUG_PORT"
+                  shift
+              # Bracketed IPv6, optional port
+              elif echo "$2" | grep -Eq '^\[[0-9A-Fa-f:]+\](:[0-9]+)?$'; then
+                  DEBUG_ADDRESS="$2"
+                  shift
+              # IPv4 or hostname:port
+              elif echo "$2" | grep -Eq '^[^:]+:[0-9]+$'; then
+                  DEBUG_ADDRESS="$2"
+                  shift
```

Comment:
It seems like there are three cases:

```suggestion
              # just a port
              if echo "$2" | grep -Eq '^[0-9]+$'; then
                  DEBUG_ADDRESS="0.0.0.0:$2"
                  shift
              # ip without a port - add the default port
              elif echo "$2" | grep -Eq '([^:]+)|(\[[0-9A-Fa-f:]+\])'; then
                  DEBUG_ADDRESS="$2:$DEBUG_PORT"
                  shift
              else
                  DEBUG_ADDRESS="$2"
                  shift
```

You then wouldn't need the check below for defaulting the address.

Also shouldn't debug address default to something like:

`DEBUG_ADDRESS="0.0.0.0:$DEBUG_PORT"`

Reply:
Added another commit for handling corner case inputs such as:

--debug [::0]
--debug ::0